### ISSUE TITLE
General Issue: Clean-up Round 2

### ISSUE THREAD

**Issue by bundfussr** [pharmaverse/admiral#1904]
### Background Information

To achieve a "stable" admiral version we need a clean-up of admiral, i.e.,
- avoid that functionality of functions overlaps
- name function and arguments consistently
- close gaps in functionality

This issue is intended to collect functions which should be revised and identify gaps.

@pharmaverse/admiral , please feel free to add.

**Comment by bundfussr** [comment]
From @kaz462 :

There are four functions for deriving extreme event/record as a new parameter, the function names are a bit confusing to me when I try to compare them: `_record(s)`/`_event`, with/without `_param`


  | Data source:      one or multiple | Extreme is ordered by - | filter/dataset source argument name
-- | -- | -- | -- 
1.[`derive_extreme_records`](https://pharmaverse.github.io/admiral/cran-release/reference/derive_extreme_records.html)| one | `order` argument | `filter`/ |
2.[`derive_extreme_event`](https://pharmaverse.github.io/admiral/cran-release/reference/derive_extreme_event.html)  | one | the order of `events` entered and the `order` argument |
3.[`derive_param_extreme_event`](https://pharmaverse.github.io/admiral/cran-release/reference/derive_param_extreme_event.html) | one  |`order` argument| `filter_source`/`dataset_source`
4.`derive_param_extreme_record` | multiple   | the order of `sources` entered and the `order` argument   | /`source_datasets` |



* What do you think of combining some functions? e.g., 

     * function 3 is a special case of function 2 (has been combined through https://github.com/pharmaverse/admiral/pull/1882)

     * function 4 is an extension of function 2 by allowing multiple data sources  

* Should we make some argument name consisnent across those functions?

**Comment by bundfussr** [comment]
Should all flag functions provide `true_vlue` and `false_value` argument?

**Comment by bundfussr** [comment]
`derive_summary_records()` should provide the `dataset_add` argument.
Consider the following use case.
```
adae <- adae %>%
  mutate(AVAL = 1)

adsaftte_aetot <- derive_summary_records(
  adae,
  filter = ANL01FL == "Y",
  set_values_to = exprs(
    PARAMCD = "AETOT1",
    PARAM = "Number of occurrences of any adverse event",
    PARCAT1 = "Total Occurrences",
    ANL01FL = "Y"
  ),    
  by_vars = exprs(STUDYID, USUBJID),
  analysis_var = AVAL,
  summary_fun = sum
) %>%
  filter(PARAMCD == "AETOT1") %>%
  select(STUDYID, USUBJID, PARAMCD, PARAM, PARCAT1, AVAL, ANL01FL)
```
The `filter()` and `select()` call at the end could be omitted if the `dataset_add` argument is available.

**Comment by bundfussr** [comment]
The arguments of `derive_vars_joined()`, `derive_var_joined_exist_flag()`, and `filter_joined()` should be made consistent. For example, all functions should provide the `dataset_add` argument.

**Comment by zdz2101** [comment]
@bms63 should we actually turn this into discussion with the respective individual pieces of consolidation as their own issues?

**Comment by bms63** [comment]
Yes lets move to discussion - also makes it easier to follow threads
